test/client/partialPayments.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-explicit-any -- required for formatting transactions */
+import { expect } from 'chai'
+
+import type { TransactionStream } from '../../src'
+import rippled from '../fixtures/rippled'
+import { setupClient, teardownClient } from '../setupClient'
+
+const partialPaymentIOU = rippled.partial_payments.iou
+const partialPaymentXRP = rippled.partial_payments.xrp
+
+describe('client handling of tfPartialPayments', function () {
+  beforeEach(setupClient)
+  afterEach(teardownClient)
+
+  it('Tx with no tfPartialPayment', async function () {
+    this.mockRippled.addResponse('tx', rippled.tx.Payment)
+    const resp = await this.client.request({ command: 'tx' })
+
+    expect(resp.warnings).to.equal(undefined)
+  })
+
+  it('Tx with IOU tfPartialPayment', async function () {
+    const mockResponse = { ...rippled.tx.Payment, result: partialPaymentIOU }
+    this.mockRippled.addResponse('tx', mockResponse)
+    const resp = await this.client.request({ command: 'tx' })
+
+    expect(resp.warnings).to.deep.equal([
+      {
+        id: 2001,
+        message: 'This response contains a Partial Payment',
+      },
+    ])
+  })
+
+  it('Tx with XRP tfPartialPayment', async function () {
+    const mockResponse = { ...rippled.tx.Payment, result: partialPaymentXRP }
+    this.mockRippled.addResponse('tx', mockResponse)
+    const resp = await this.client.request({ command: 'tx' })
+
+    expect(resp.warnings).to.deep.equal([
+      {
+        id: 2001,
+        message: 'This response contains a Partial Payment',
+      },
+    ])
+  })
+
+  it('account_tx with no tfPartialPayment', async function () {
+    this.mockRippled.addResponse('account_tx', rippled.account_tx.normal)
+    const resp = await this.client.request({ command: 'account_tx' })
+
+    expect(resp.warnings).to.equal(undefined)
+  })
+
+  it('account_tx with IOU tfPartialPayment', async function () {
+    const partial = {
+      ...rippled.tx.Payment,
+      result: partialPaymentIOU,
+    }
+    const mockResponse = rippled.account_tx.normal
+    mockResponse.result.transactions.push({
+      tx: partial.result,
+      meta: partial.result.meta,
+    } as any)
+
+    this.mockRippled.addResponse('account_tx', mockResponse)
+    const resp = await this.client.request({ command: 'account_tx' })
```

Comment:
ditto

Reply:
I can add it, but it doesn't effect anything.